src/main/scala/com/kinja/soy/Soy.scala

```diff
@@ -63,4 +63,7 @@ object Soy {
    * Generates a SoyMapWrites[T] for a given case class T.
    */
   def writes[T]: SoyMapWrites[T] = macro SoyMacroImpl.writesImpl[T]
+
+  def typedWrites[T](typeValue: String): SoyMapWrites[T] = macro SoyMacroImpl.typedWritesImpl[T]
```

Comment:
Our random one-offs are not used that way but we want to standardize this. We'll of course have to migrate the places where we're sending a non-standard `type` field but in the long run the consistency across the code-base and reduced possibility space will be worth the couple of minutes of work changing the template to check `ScheduledRepost` instead of `repost`.